File: 75. Sort Colors/optimised.ts
function sortColors(nums: number[]): void {
	const count = [0, 0, 0];

	for (const num of nums) {
		count[num]++;
	}

	let i = 0;
	for (let c = 0; c < 3; c++) {
		while (count[c] > 0) {
			nums[i] = c;
			i++;
			count[c]--;
		}
	}
};

const nums = [2,0,2,1,1,0];
sortColors(nums);
console.log(nums);

File: 75. Sort Colors/optimised.py
#!/usr/bin/env python

from typing import List

def sortColors(nums: List[int]) -> None:
    count = [0, 0, 0]

    for num in nums:
        count[num] += 1

    i = 0

    for c in range(3):
        while count[c] > 0:
            nums[i] = c
            i += 1
            count[c] -= 1

nums = [2,0,2,1,1,0]
sortColors(nums)
print(nums)
